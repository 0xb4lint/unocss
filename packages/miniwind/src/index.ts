export * from '@miniwind/core'
export * from '@miniwind/preset-default'
export * from '@miniwind/preset-attributify'
