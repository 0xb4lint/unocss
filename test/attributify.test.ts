import { createGenerator, presetDefault, variantAttributify, extractorAttributify, presetAttributify } from 'miniwind'

describe('attributify', () => {
  const code = `
<button 
  bg="blue-400 hover:blue-500 dark:!blue-500 dark:hover:blue-600"
  text="sm white"
  flex="~ col"
  p="t-2"
  pt="2"
  border="rounded-xl"
  :font="condition ? 'mono' : 'sans'"
  v-bind:p="y-2 x-4"
  border="2 rounded blue-200"
  mw-children="m-auto"
>
  Button
</button>
`

  const extract = extractorAttributify(code, '')

  test('extractor', async() => {
    expect(await extract).toMatchSnapshot()
  })

  test('variant', async() => {
    const variant = variantAttributify()
    expect(Array.from(await extract || []).map(i => variant.match(i, {} as any))).toMatchSnapshot()
  })

  test('generate', async() => {
    const generate = createGenerator({
      presets: [
        presetAttributify(),
        presetDefault(),
      ],
    })
    const { css } = await generate(code)
    expect(css).toMatchSnapshot()
  })
})
