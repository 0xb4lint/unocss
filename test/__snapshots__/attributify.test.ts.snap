// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attributify extractor 1`] = `
Set {
  "[bg~=\\"blue-400\\"]",
  "[bg~=\\"hover:blue-500\\"]",
  "[bg~=\\"dark:!blue-500\\"]",
  "[bg~=\\"dark:hover:blue-600\\"]",
  "[text~=\\"sm\\"]",
  "[text~=\\"white\\"]",
  "[flex~=\\"~\\"]",
  "[flex~=\\"col\\"]",
  "[p~=\\"t-2\\"]",
  "[pt~=\\"2\\"]",
  "[border~=\\"rounded-xl\\"]",
  "[font~=\\"condition\\"]",
  "[font~=\\"?\\"]",
  "[font~=\\"mono\\"]",
  "[font~=\\":\\"]",
  "[font~=\\"sans\\"]",
  "[p~=\\"y-2\\"]",
  "[p~=\\"x-4\\"]",
  "[border~=\\"2\\"]",
  "[border~=\\"rounded\\"]",
  "[border~=\\"blue-200\\"]",
  "[mw-children~=\\"m-auto\\"]",
  "[hover:blue-500=\\"\\"]",
  "[rounded=\\"\\"]",
  "[pt2=\\"\\"]",
  "[inline-block=\\"\\"]",
}
`;

exports[`attributify generate 1`] = `
"[p~=\\"y-2\\"]{padding-top:0.5rem;padding-bottom:0.5rem;}
[p~=\\"x-4\\"]{padding-left:1rem;padding-right:1rem;}
[pt2=\\"\\"],[pt~=\\"\\\\32 \\"],[p~=\\"t-2\\"],.pt2{padding-top:0.5rem;}
[mw-children~=\\"m-auto\\"] > *,.m-auto{margin:auto;}
[inline-block=\\"\\"],.inline-block{display:inline-block;}
[bg~=\\"blue-400\\"]{--mw-bg-opacity:1;background-color:rgba(96,165,250,var(--mw-bg-opacity));}
[bg~=\\"hover\\\\:blue-500\\"]:hover{--mw-bg-opacity:1;background-color:rgba(59,130,246,var(--mw-bg-opacity));}
.dark [bg~=\\"dark\\\\:\\\\!blue-500\\"]{--mw-bg-opacity:1 !important;background-color:rgba(59,130,246,var(--mw-bg-opacity)) !important;}
.dark [bg~=\\"dark\\\\:hover\\\\:blue-600\\"]:hover{--mw-bg-opacity:1;background-color:rgba(37,99,235,var(--mw-bg-opacity));}
[border~=\\"\\\\32 \\"]{border-width:2px;border-style:solid;}
[border~=\\"blue-200\\"]{--mw-border-opacity:1;border-color:rgba(191,219,254,var(--mw-border-opacity));}
[border~=\\"rounded-xl\\"],.rounded-xl{border-radius:0.75rem;}
[rounded=\\"\\"],[border~=\\"rounded\\"],.rounded{border-radius:0.25rem;}
[font~=\\"mono\\"]{font-family:ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,\\"Liberation Mono\\",\\"Courier New\\",monospace;}
[font~=\\"sans\\"]{font-family:ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont,\\"Segoe UI\\",Roboto,\\"Helvetica Neue\\",Arial,\\"Noto Sans\\",sans-serif,\\"Apple Color Emoji\\",\\"Segoe UI Emoji\\",\\"Segoe UI Symbol\\",\\"Noto Color Emoji\\";}
[text~=\\"sm\\"]{font-size:0.875rem;line-height:1.25rem;}
[text~=\\"white\\"]{--mw-text-opacity:1;color:rgba(255,255,255,var(--mw-text-opacity));}
[flex~=\\"col\\"]{flex-direction:column;}
[flex~=\\"\\\\~\\"]{display:flex;}"
`;

exports[`attributify variant 1`] = `
Array [
  "bg-blue-400",
  "hover:bg-blue-500",
  "dark:!bg-blue-500",
  "dark:hover:bg-blue-600",
  "text-sm",
  "text-white",
  "flex",
  "flex-col",
  "p-t-2",
  "pt-2",
  "border-rounded-xl",
  "font-condition",
  "font-?",
  "font-mono",
  "font-:",
  "font-sans",
  "p-y-2",
  "p-x-4",
  "border-2",
  "border-rounded",
  "border-blue-200",
  "children-m-auto",
  "hover:blue-500",
  "rounded",
  "pt2",
  "inline-block",
]
`;
